pkg/metrics/metrics.go

```diff
@@ -0,0 +1,190 @@
+	)
+	// MetricJiraIssueCreateFailures is the number of compliance events that failed to be created as Jira issues
+	MetricJiraIssueCreateFailures = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name:        "compliance_audit_router_jira_issue_create_failures",
+		Help:        "Number of Jira issues that failed to be created",
+		ConstLabels: CARPrometheusLabels},
+		[]string{"uuid", "process"},
+	)
+
+	// JIRA WEBHOOK PROCESSING
+
+	// MetricJiraWebhookReceived is the number of Jira notification webhooks received
+	// Jira notification webhooks are received when a Jira issue is updated
+	MetricJiraWebhookReceived = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name:        "compliance_audit_router_jira_webhook_received",
+		Help:        "Number of Jira notification webhooks received",
+		ConstLabels: CARPrometheusLabels},
+		[]string{"uuid", "process"},
+	)
+	// MetricJiraWebhookProcessFailures is the number of Jira notification webhooks that failed to be processed
+	MetricJiraWebhookProcessFailures = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name:        "compliance_audit_router_jira_webhook_process_failures",
+		Help:        "Number of Jira notification webhooks notifications that failed to be processed",
+		ConstLabels: CARPrometheusLabels},
+		[]string{"uuid", "process"},
+	)
+	// MetricJiraIssueUpdateFailures is the number of failures updating issues based on received webhook events
+	MetricJiraIssueUpdateFailures = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name:        "compliance_audit_router_jira_issue_update_failures",
+		Help:        "Number of Jira issues that failed to be updated based on received webhook",
+		ConstLabels: CARPrometheusLabels},
+		[]string{"uuid", "process"},
+	)
+
+	// LDAP LOOKUP PROCESSING
+
+	// MetricLDAPLookupFailures is the number of LDAP lookups that failed
+	// Other LDAP metrics are not currently tracked, as they are not integral to the process
+	MetricLDAPLookupFailures = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name:        "compliance_audit_router_ldap_lookup_failures",
+		Help:        "Number of LDAP lookups that failed",
+		ConstLabels: CARPrometheusLabels},
+		[]string{"uuid", "process"},
+	)
+
+	// HTTP RESPONSES TO CLIENTS
+
+	// MetricHTTPErrors is the number of HTTP errors returned by the application
+	MetricHTTPErrors = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name:        "compliance_audit_router_http_errors",
+		Help:        "Number of HTTP errors returned by the application",
+		ConstLabels: CARPrometheusLabels},
+		[]string{"code", "uuid", "process"},
+	)
+	// MetricHTTPSuccesses is the number of HTTP successes returned by the application
+	MetricHTTPSuccesses = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name:        "compliance_audit_router_http_successes",
+		Help:        "Number of HTTP successes returned by the application",
+		ConstLabels: CARPrometheusLabels},
+		[]string{"code", "uuid", "process"},
+	)
```

Comment:
What do you think about returning HTTP code as a label of this metric? Then you could slice and dice the data as you wanted with PromQL?

Reply:
Great idea - replaced these two with `MetricHTTPResponses` in the code, and can use the `[]string{"code"...` bit as the label for the response code.